WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2828,16 +2835,38 @@ static GSourceFuncs _handlerIntervention =
                     webkit_memory_pressure_settings_free(memoryPressureSettings);
                 }
 #endif
+
+#if WEBKIT_CHECK_VERSION(2, 42, 0)
+                double originStorageRatio = -1.0;    // -1.0 means WebKit will use the default quota (1GB)
+                if (_config.OriginStorageRatio.IsSet() && _config.OriginStorageRatio.Value() != 0) {
+                    originStorageRatio = static_cast<double>(_config.OriginStorageRatio.Value());
+                }
+
+                double totalStorageRatio = -1.0;    // -1.0 means there's no limit for the total storage
+                if (_config.TotalStorageRatio.IsSet() && _config.TotalStorageRatio.Value() != 0) {
+                    totalStorageRatio = static_cast<double>(_config.TotalStorageRatio.Value());
+                }
+
                 auto* websiteDataManager = webkit_website_data_manager_new(
                     "local-storage-directory", wpeStoragePath,
                     "disk-cache-directory", wpeDiskCachePath,
                     "local-storage-quota", localStorageDatabaseQuotaInBytes,
                     "indexeddb-directory", indexedDBPath,
-                    "per-origin-storage-quota", indexedDBSizeBytes,
+                    "origin-storage-ratio", originStorageRatio,
+                    "total-storage-ratio", totalStorageRatio,
+                    "base-data-directory", _service->PersistentPath(),
+                    "base-cache-directory", _service->VolatilePath(),
+                     nullptr);
+                g_free(indexedDBPath);
+#else
```

Comment:
This is the 2.38 and earlier code-block. When this PR created there was no potential usage of indexdb. But with a prime app, there might be usage of indexdb. 

@emutavchi, to not change the config options for coming releases, shall we use same originstorageratio and totalstorageratio which are valid from 2.42 on (we can pass the total ratio empty as we can calculate from persistent path) from JSON config file? 

we are working on a document to explain all storage apis, but in the meantime you can find basic info from this ticket: https://github.com/WebPlatformForEmbedded/WPEWebKit/issues/1251#issuecomment-1847138858

If it is ok for you for 2.38 it will be like this:
auto* websiteDataManager = webkit_website_data_manager_new(
                     "local-storage-directory", wpeStoragePath,
                     "disk-cache-directory", wpeDiskCachePath,
                     "local-storage-quota", localStorageDatabaseQuotaInBytes,
                     "indexeddb-directory", indexedDBPath,
                     "per-origin-storage-quota", **CALCULATED FROM originStorageRatio**
                     "base-data-directory", _service->PersistentPath(),
                     "base-cache-directory", _service->VolatilePath(),
                      nullptr);

Reply:
FYI @VeithMetro , @MFransen69

Reply:
@modeveci I updated all *.config files which contained a `indexeddbsize` setting.

Two questions:

1. JSPP.config does not have `indexeddbsize` settings, but JSPP.conf.in does, should we add it to .config as well?
2. Should the `per-origin-storage-quota` be calculated as `LocalStorageSize` times `originStorageRatio` (in a separate `#if WEBKIT_CHECK_VERSION(2, 38, 0)`)?

Reply:
Hi @VeithMetro ,
1. Yes, you can add indexdbsize to JSPP.config, your PR adds config options to others as well. 
2. No, LocalStorageSize is different. You can calculate with "base-data-directory" which is persistentPath.

Reply:
I added indexdbsize to JSPP.config. But I am not completely sure about the `per-origin-storage-quota`.

Do we want to calculate the maximum possible size of a directory under `base-data-directory`, which like you said is `persistentPath`? In that case, if I am not mistaken, the `per-origin-storage-quota` would be equal to all free space on a disk which contains the `persistentPath' times `originStorageRatio`. Is this what we are looking for?

If so, I don't think there is a way to check a maximum possible size of a directory in Thunder, and I don't think there is a limit set on the maximum size of `persistentPath` in Thunder, so it is effectively a size of the whole mounted disk.

Reply:
RDK manifests set rdkservices' persistent path as /opt/persistent/rdkservices. And you are right that is basically pointing whole /dev/root filesystem. I think it is another discussion whether we will or want to give a maximum persistent storage size. 

So far,  to get the size of persistentPath, we can use Core::File::Size(theNameofDirectory)

**From 2.42 on, we can tell webkit to use as "total-storage-ratio" of persistent path for all domains. That will control and manage eviction algorithms.** 
E.g:
```
base-data-directory (persistent-path): 1GB
total-storage-ratio (0.1 of base-data-directory): 100MB
origin-storage-ratio(0.01 of base-data-directory): 10MB
```

**For 2.38, we can again fetch the size of persistent path by calling Core::File:Size API**
and e.g:
```
base-data-directory (persistent-path): 1GB
per-origin-storage-quota = 10MB = originStorageRatio(0.01) X persistentPath (1GB)
```

If @emutavchi and @MFransen69 are ok with this approach, then we can make changes.

Reply:
@modeveci @VeithMetro /opt/persistent could be (and usually is) a separate partition. Perhaps 'WPEFramework::Core::Partition' can give you the size of partition and free space available there.

Reply:
Thank you for your feedback @emutavchi. to be more generic it is better to check the size with Core::File::Size. if the /opt/persistent is a partition it will be the size of that partition. if it is not a separate partition, It will be the size of /dev/root. With that, it can work for other platforms (including development platforms) as well.

Reply:
@modeveci is it now confirmed that we want to use `Core::File::Size` to get the free space available under `base-data-directory` path and then calculate the `per-origin-storage-quota`? 😄

Reply:
Let's start to use ::Size as it is generic. Please prepare your changes.